tests/base/src/test/java/org/keycloak/tests/admin/event/LoginEventsTest.java

```diff
@@ -0,0 +1,255 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.keycloak.tests.admin.event;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.events.Event;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.EventType;
+import org.keycloak.representations.idm.EventRepresentation;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.oauth.OAuthClient;
+import org.keycloak.testframework.oauth.annotations.InjectOAuthClient;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.RealmConfig;
+import org.keycloak.testframework.realm.RealmConfigBuilder;
+import org.keycloak.testframework.remote.runonserver.InjectRunOnServer;
+import org.keycloak.testframework.remote.runonserver.RunOnServerClient;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test getting and filtering login-related events.
+ *
+ * @author Stan Silvert ssilvert@redhat.com (C) 2016 Red Hat Inc.
+ */
+@KeycloakIntegrationTest
+public class LoginEventsTest {
+
+    @InjectRealm(config = LoginEventsRealmConfig.class)
+    ManagedRealm managedRealm;
+
+    @InjectOAuthClient
+    OAuthClient oAuthClient;
+
+    @InjectRunOnServer
+    RunOnServerClient runOnServerClient;
+
+    @BeforeEach
+    public void init() {
+        managedRealm.admin().clearEvents();
+    }
+
+    private List<EventRepresentation> events() {
+        return managedRealm.admin().getEvents();
+    }
+
+    private void badLogin() {
+        oAuthClient.doLogin("bad", "user");
+    }
+
+    private void pause(int seconds) {
+        try {
+            Thread.sleep(seconds * 1000L);
+        } catch (InterruptedException e) {
+            Assertions.fail(e.getMessage());
+        }
+    }
+
+    // todo Where should the isUUID function be?
+
+    /**
+     * @see org.keycloak.testsuite.AssertEvents#isUUID()
+     */
+    private static boolean isUUID(String item) {
+        return 36 == item.length() && item.charAt(8) == '-' && item.charAt(13) == '-' && item.charAt(18) == '-' && item.charAt(23) == '-';
+    }
+
+    @Test
+    public void eventAttributesTest() {
+        badLogin();
+        List<EventRepresentation> events = events();
+        Assertions.assertEquals(1, events.size());
+        EventRepresentation event = events.get(0);
+        Assertions.assertTrue(isUUID(event.getId()));
```

Comment:
This doesn't make much sense. This should really be something like:

`event.id(Matcher.isUUID())`

We could potentially just also bake directly into AdminEventAssertion that it verifies ID is set and that it's a UUID.